test/AccountManager.test.ts

```diff
@@ -0,0 +1,292 @@
+          const actualState = accountManager.isDeployRequest(deployRequest);
+
+          expect(actualState).to.be.true;
+        });
+
+        it('should return `false` if arg is a relay request', function () {
+          const actualState = accountManager.isDeployRequest(relayRequest);
+
+          expect(actualState).to.be.false;
+        });
+      });
+
+      describe('getAccounts', function () {
+        it('should return list of local account addresses from', function () {
+          const wallets: Wallet[] = [
+            Wallet.createRandom(),
+            Wallet.createRandom(),
+            Wallet.createRandom(),
+          ];
+          const expectedAddresses: string[] = wallets.map((w) => w.address);
+
+          (
+            accountManager as unknown as {
+              _accounts: Wallet[];
+            }
+          )._accounts = wallets;
+
+          expect(accountManager.getAccounts()).to.deep.equal(expectedAddresses);
+        });
+      });
+
+      describe('sign', function () {
+        it('should check the type of request', async function () {
+          const stubIsDeployRequest = sandbox.stub(
+            accountManager,
+            'isDeployRequest'
+          );
+          await accountManager.sign(relayRequest).catch(() => undefined);
+
+          expect(stubIsDeployRequest).to.be.calledWith(relayRequest);
+        });
+
+        it('should construct typed message', async function () {
+          (accountManager as unknown as { _accounts: Wallet[] })._accounts.push(
+            Wallet.createRandom()
+          );
+          const expectedMessage = {
+            types: {
+              EIP712Domain: typedDataUtils.eip712DomainType,
+              RelayRequest: typedDataUtils.relayRequestType,
+              RelayData: typedDataUtils.relayDataType,
+            },
+            primaryType: 'RelayRequest',
+            domain: typedDataUtils.getDomainSeparator(
+              relayRequest.relayData.callForwarder as string,
+              accountManager.chainId
+            ),
+            message: {
+              ...relayRequest.request,
+              relayData: relayRequest.relayData,
+            },
+          };
+          const stubRecoverSignature = sandbox
+            .stub(
+              accountManager as unknown as {
+                _recoverSignature: () => string;
+              },
+              '_recoverSignature'
+            )
+            .callsFake(() => relayRequest.request.from.toString());
+          await accountManager.sign(relayRequest);
+
+          expect(stubRecoverSignature).to.be.calledWith(
+            expectedMessage,
+            sandbox.match.any
+          );
+        });
+
+        it('should verify the request originator address against recovered typed signature', async function () {
+          sandbox
+            .stub(
+              accountManager as unknown as {
+                _recoverSignature: () => string;
```

Comment:
should we use this approach for stubbing private keys?

Reply:
no alternative os stubbing privates, on the unit tests that I have done, the private method was execute since I did not stub it.

Reply:
ok then I don't know what the status of this comment is now....

Reply:
@jurajpiar I think that the point here is: if I stub the private methods, how am I going to test them? Should we create unit tests for private methods? I'd say no unless strictly required, so we should probably execute the private methods without stubbing them, shouldn't we?

Reply:
not if the private method doesn't contribute to the test case

Reply:
>Also another issue is that all this private method does is calling an dependency. Now, normally the dependency should be stubbed, but I was unable to stub sigUtil directly, so I wrapped it in a method. Should you find a better solution I will be more than happy to change it.

got it, if you couldn't stub the dependency, it makes sense to me